pkgs/development/python-modules/trio/default.nix

```diff
@@ -13,19 +13,45 @@
 , jedi
 , astor
 , yapf
+, coreutils
 }:
 
 buildPythonPackage rec {
   pname = "trio";
-  version = "0.19.0";
+  version = "0.20.0";
   disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "895e318e5ec5e8cea9f60b473b6edb95b215e82d99556a03eb2d20c5e027efe1";
+    sha256 = "sha256-ZwpS0xFdDoeeGsg4pOuZmvMvhYFj46cE/kg53ipnYHA=";
   };
 
-  checkInputs = [ astor pytestCheckHook pyopenssl trustme jedi yapf ];
+  propagatedBuildInputs = [
+    attrs
+    sortedcontainers
+    async_generator
+    idna
+    outcome
+    sniffio
+  ] ++ lib.optionals (pythonOlder "3.7") [ contextvars ];
+
+  # tests are failing on Darwin
+  doCheck = !stdenv.isDarwin;
+
+  checkInputs = [
+    astor
+    jedi
+    pyopenssl
+    pytestCheckHook
+    trustme
+    yapf
+  ];
+
+  preCheck = ''
```

Comment:
```suggestion
  postPatch = ''
```

Reply:
Aha, what's the benefit?

Reply:
That all code patching is done in patches or postPatch and this isn't changing after building, installing or checking. Doing patching in multiple places is getting really messy the more things are changed especially when it is done in multiple phases.

Reply:
Agree to disagree.